cmd/daemon/api_server.go

```diff
@@ -251,13 +266,23 @@ func (s *ApiServer) handleRequest(req ApiRequest, w http.ResponseWriter) {
 	req.resp = make(chan apiResponse, 1)
 	s.requests <- req
 	resp := <-req.resp
-	if errors.Is(resp.err, ErrNoSession) {
-		w.WriteHeader(http.StatusNoContent)
-		return
-	} else if resp.err != nil {
-		log.WithError(resp.err).Error("failed handling status request")
-		w.WriteHeader(http.StatusInternalServerError)
-		return
+
+	if resp.err != nil {
+		switch resp.err {
+		case ErrNoSession:
+			w.WriteHeader(http.StatusNoContent)
+			return
+		case ErrNotFound:
+			w.WriteHeader(http.StatusNotFound)
+			return
+		case ErrNotFound:
+			w.WriteHeader(http.StatusNoContent)
+			return
```

Comment:
I think this is supposed to be `ErrMethodNotAllowed` and `StatusMethodNotAllowed`.